src/main/osd/osd_warnings.h

```diff
@@ -20,11 +20,14 @@
 
 #pragma once
 
-#define OSD_WARNINGS_MAX_SIZE 12
+#include <stdbool.h>
+#include <stdint.h>
+
+#define OSD_WARNINGS_MAX_SIZE 31
```

Comment:
This needs to be adapted: https://github.com/betaflight/betaflight/blob/master/src/main/osd/osd_warnings.c#L287

Reply:
Did a mockup in online compiler for testing it results in

```bash
          1         2         3
0123456789012345678901234567890
            ESC1RTC
```

As before we had 

```bash
          1         2         3
0123456789012345678901234567890
  ESC1RTC
```

This holds true for 30 columns OSD

Reply:
With 8 motors
```
          1         2         3
0123456789012345678901234567890
          ESC1RTC5678
```

Reply:
<details>
<summary><b>mockup</b></summary>

```bash
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <stdlib.h>

#define OSD_WARNINGS_MAX_SIZE 31
#define OSD_FORMAT_MESSAGE_BUFFER_SIZE (OSD_WARNINGS_MAX_SIZE + 1)

int getMotorCount(void) {
    return 8;
}

int main()
{
    char escWarningMsg[OSD_FORMAT_MESSAGE_BUFFER_SIZE];
    unsigned pos = 0;

    const char *title = "ESC";

    // center justify message

    pos = (OSD_WARNINGS_MAX_SIZE - (strlen(title) + getMotorCount())) / 2;
    memset(escWarningMsg, ' ', pos);
    
    printf("\nPos = %d", pos);

    strncpy(escWarningMsg + pos, title, OSD_FORMAT_MESSAGE_BUFFER_SIZE - pos - 1);
    pos += strlen(title);

    printf("\nWarningMsg = %s", escWarningMsg);

    unsigned i = 0;
    unsigned escWarningCount = 0;

    for (unsigned i = 0; i < getMotorCount() && pos < OSD_FORMAT_MESSAGE_BUFFER_SIZE - 1; i++) {
        printf("\nPos %d", pos);

        const char motorNumber = '1' + i;
        // if everything is OK just display motor number else R, T or C
        char warnFlag = motorNumber;
        if (i == 1) {
            warnFlag = 'R';
        }
        if (i == 2) {
            warnFlag = 'T';
        }
        if (i == 3) {
            warnFlag = 'C';
        }

        escWarningMsg[pos++] = warnFlag;

        if (warnFlag != motorNumber) {
            escWarningCount++;
        }
    }

    escWarningMsg[pos] = '\0';

    if (escWarningCount > 0) {
        printf("\n          1         2         3");
        printf("\n0123456789012345678901234567890");
        printf("\n%s", escWarningMsg);
        return 0;
    }

    return 0;
}
```

</details>

Reply:
It's fine if OSD layout allows all 30 character to be printed. But if old layout assumes 11 character warning, centering code may make the layout pretty ugly

Reply:
@haslinghuis : https://github.com/betaflight/betaflight/commit/cf6b3639e29ca82b3e9e79beb7e9f00ef175c590

This is problem in BF - some code is simply low quality, some code was written under different circumstances. The code is slowly modified and it is often used as template for new one. In both cases the code slowly rots away.

From time to time, it is necessary to do deeper cut and adapt code for new situation.

And it helps to start from simple, easy to understand code. That is why I'm trying to get new PRs to some basic quality - 'works for me' and 'flight tested' isn't good approach for project as big as BF ...

Reply:
- Agree code quality and code are a mixed bag. Really grateful for all efforts improving code quality and superb reviews.  All in for standards and guidelines.
- Noticed `MAX_SUPPORTED_MOTORS` used to be 12 so when we would decide to revert that would break size again.

Reply:
> * Noticed `MAX_SUPPORTED_MOTORS` used to be 12 so when we would decide to revert that would break size again.

Actually there is no explicit or documented limit for number of motors, target.h can theoretically set MAX_SUPPORTED_MOTORS  to almost anything. But assuming 8 is IMO reasonable and can be documented ('check all relevant code if MAX_SUPPORTED_MOTORS is increased above 8')

Correct way is to check buffer limit where relevant (you can reasonably expect that the size won't be decreased, so most constant strings are fine now with 31bytes buffer). And simply truncate output when there are too many motors.

Reply:
Perhaps just

```bash
#ifdef USE_ESC_SENSOR
    // Show warning if we lose motor output, the ESC is overheating or excessive current draw
    if (featureIsEnabled(FEATURE_ESC_SENSOR) && osdWarnGetState(OSD_WARNING_ESC_FAIL) && ARMING_FLAG(ARMED)) {
        int index = 0;
        unsigned escWarningCount = 0;

        warningText[index++] = 'E';
        warningText[index++] = 'S';
        warningText[index++] = 'C';

        for (unsigned i = 0; i < getMotorCount(); i++) {
            escSensorData_t *escData = getEscSensorData(i);
            const char motorNumber = '1' + i;
            // if everything is OK just display motor number else R, T or C
            char warnFlag = motorNumber;

            if (osdConfig()->esc_rpm_alarm != ESC_RPM_ALARM_OFF && (uint32_t)erpmToRpm(escData->rpm) <= (uint32_t)osdConfig()->esc_rpm_alarm) {
                warnFlag = 'R';
            }
            if (osdConfig()->esc_temp_alarm != ESC_TEMP_ALARM_OFF && escData->temperature >= osdConfig()->esc_temp_alarm) {
                warnFlag = 'T';
            }
            if (osdConfig()->esc_current_alarm != ESC_CURRENT_ALARM_OFF && escData->current >= osdConfig()->esc_current_alarm) {
                warnFlag = 'C';
            }

            warningText[index++] = warnFlag;

            if (warnFlag != motorNumber) {
                escWarningCount++;
            }
        }

        if (escWarningCount > 0) {
            warningText[index] = 0;
            *displayAttr = DISPLAYPORT_SEVERITY_WARNING;
            *blinking = true;
            return;
        } else {
            warningText[0] = 0;
        }
    }
#endif // USE_ESC_SENSOR

```

Reply:
I update https://github.com/haslinghuis/betaflight/pull/6/files with one possible implementation